src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -105,52 +129,340 @@ class boss_gluth : public CreatureScript
+                if (timer>1600 && me->GetExactDist2d(gluth) > 10.0 && me->CanFreeMove()) // it takes about 1600 ms for the animation to cycle. This way, the animation looks relatively smooth.
                 {
-                    me->Kill(me->GetVictim());
-                    me->ModifyHealth(int32(me->CountPctFromMaxHealth(5)));
+                    me->GetMotionMaster()->MovePoint(0, gluth->GetPosition()); // isn't dynamic. So, to take into account Gluth's movement, it must be called periodicly.
+                    //me->GetMotionMaster()->MoveFollow(gluth, 0, 0); // the zombies will stand still too far away from Gluth at the end of their movement. Also doesnt seem to work with SetWalk(true)
+                    timer = 0;
                 }
+
+                if (me->GetExactDist2d(gluth) <= 10.0)
+                    me->StopMoving();
             }
-            else
+            else if (state == STATE_ZOMBIE_NORMAL)
                 DoMeleeAttackIfReady();
         }
+
+        void SetData(uint32 id, uint32 value) override
+        {
+            if (id == DATA_ZOMBIE_STATE) // change of state
+            {
+                state = value;
+                if (value == STATE_ZOMBIE_DECIMATED)
+                {
+                    me->SetReactState(ReactStates::REACT_PASSIVE);
+                    me->AttackStop();
+                    me->SetTarget(ObjectGuid::Empty);
+                    // at this point, the zombie should be non attacking and non moving.
+
+                    me->SetWalk(true); // it doesnt seem to work with MoveFollow() (but it does work with MovePoint()). 
+                    //me->SetSpeed(UnitMoveType::MOVE_RUN, baseMoveSpeed[UnitMoveType::MOVE_WALK] / baseMoveSpeed[UnitMoveType::MOVE_RUN]); // Hack to stay in run mode but with walk speed in case using MoveFollow()
+
+                    timer = 1000;
+                }
+                else if (value == STATE_ZOMBIE_TOBE_EATEN)
+                {
+                    // loose aggro behavior
+                    me->SetReactState(ReactStates::REACT_PASSIVE);
+                    me->AttackStop();
+                    me->SetTarget(ObjectGuid::Empty);
+
+                    // and forced to stand still
+                    me->StopMoving();
+                    me->GetMotionMaster()->Clear();
+                    me->GetMotionMaster()->MoveIdle();
+                    me->ApplySpellImmune(0, IMMUNITY_MECHANIC, MECHANIC_GRIP, true); // not sure if this is blizz-like but this is very convenient
+                }
+            }
+        }
+
+        uint32 GetData(uint32 index) const override
+        {
+            if (index == DATA_ZOMBIE_STATE)
+                return state;
+            return 0;
+        }
+
+    private:
+        uint32 timer;
+        uint8 state;
+        ObjectGuid GluthGUID;
     };
 
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_zombie_chowAI(creature);
```

Comment:
Use [`GetInstanceAI`](https://github.com/Treeston/TrinityCore/blob/3.3.5-naxxcleanup/src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp#L167) to make sure `me->GetInstanceScript()` is always not null. Otherwise, core crash if you add the NPC outside of an instance.